src/main/kotlin/assimp/MemoryIOWrapper.kt

```diff
@@ -32,14 +32,24 @@ class MemoryIOSystem(val buffer: ByteBuffer) : IOSystem{
 			get() = null
 
 		override fun read(): InputStream {
-			return ByteBufferBackedInputStream(buffer)
+			return ByteBufferBackedInputStream(readBytes())
 		}
 
 		override fun reader(): BufferedReader {
 			return BufferedReader(InputStreamReader(read()))
 		}
 
 		override fun parentPath(): String = ""
+
+		override val length: Long
+			get() = buffer.size.toLong()
+
+		// TODO figure out whether or not I should call dup here
+		// do multiple calls of read() on the default implementation generate input streams
+		// with positions that match each other or not?
+		// If I move duplicate back here I need to also adjust the byte order here
+		// Then I no longer need to change it in MemoryIOSystem.open
+		override fun readBytes(): ByteBuffer = buffer
```

Comment:
I would call duplicate here, then remove all other duplicates in the code while making sure the internal buffer is not used